Remove a silent error check to let it fall through into the next

therealprof opened the pull request:
The next error check is much more suitable to hangle the error due to
the error message to let the user know that something went wrong.

Fixes #7661.

Signed-off-by: Daniel Egger <daniel@eggers-club.de>

mike-scott commented:
I think coap_find_options() used to return a variety of error codes and print related errors when it was the zoap_find_options() API (now depreciated).  But, looking at coap_find_options() as it is today, there really isn't any better error code we can print to the user.  Either a parameter going into it was bad to start (EINVAL) or we hit an error during the parsing of the CoAP option data (EINVAL).

My concern would be always returning -EINVAL vs. returning the error from coap_find_options() as over time there could be changes in that code.  Either way we *ARE* getting unexpected endpoint data (too little).

There is another case where we get more than enough and we have no idea which option is valid, but that's a fix for another day.  We might change the check to <> 2.

therealprof commented:
> Either way we ARE getting unexpected endpoint data (too little).

That's kind of my point. ;) Passing the error code on doesn't make much sense to me because that could be anything and the caller of `do_registration_reply_cb` would need to know how to deal with that error code to, which is bad IMHO; it should always be transparent what might be returned and it's an `EINVAL` after all...

Of course if there's a useful different error message we could output, putting that in would also work for me.

therealprof commented:
NB: It is possible that one of the other reported bugs is actually a phantom of this problem. I haven't quite figured out in which cases the Zephyr build system decides to wipe out my `.config` and replace it with a different one based on the general and project configuration.

therealprof committed:
net: lwm2m: remove silent fail for bad endpoint data in rd_client

The next error check is much more suitable to handle the error due to
the error message which lets the user know that something went wrong.

Fixes #7661.

Signed-off-by: Daniel Egger <daniel@eggers-club.de>

therealprof commented:
@mike-scott Complaints should be addressed now

mike-scott reviewed (approved):
LGTM
